pkg/cluster/backupstorage/pv.go

```diff
@@ -0,0 +1,48 @@
+package backupstorage
+
+import (
+	"github.com/coreos/etcd-operator/pkg/spec"
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+
+	"k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+type pv struct {
+	clusterName   string
+	namespace     string
+	pvProvisioner string
+	backupPolicy  spec.BackupPolicy
+	kubecli       *unversioned.Client
+}
+
+func NewPVStorage(kc *unversioned.Client, cn, ns, pvp string, backupPolicy spec.BackupPolicy) Storage {
+	return &pv{
+		clusterName:   cn,
+		namespace:     ns,
+		pvProvisioner: pvp,
+		backupPolicy:  backupPolicy,
+		kubecli:       kc,
+	}
+}
+
+func (s *pv) Init() error {
+	return k8sutil.CreateAndWaitPVC(s.kubecli, s.clusterName, s.namespace, s.pvProvisioner, s.backupPolicy.VolumeSizeInMB)
+}
+
+func (s *pv) Clone(from string) error {
+	if s.clusterName == from {
```

Comment:
the logic of clone outside is changed to look cleaner.

See backupManager.setup()